src/Products/Viewer/Controllers/ViewerApiController.cs

```diff
@@ -521,197 +602,203 @@ private static ViewerSettings GetViewerSettings(string documentGuid, out string
-                loadDocumentEntity.SetPages(pageData);
+                xdoc.Save(pagesInfoPath);
             }
-
-            loadDocumentEntity.SetGuid(documentGuid);
-
-            return loadDocumentEntity;
         }
 
-        internal class MemoryPageStreamFactory : IPageStreamFactory
+        /// <summary>
+        /// Calculates new page rotation angle value.
+        /// </summary>
+        /// <param name="currentAngle">Current page rotation angle value.</param>
+        /// <param name="postedAngle">Posted page rotation angle value.</param>
+        /// <returns>New page rotation angle value.</returns>
+        private static int GetNewAngleValue(int currentAngle, int postedAngle)
         {
-            private readonly List<MemoryStream> _pages;
-
-            public MemoryPageStreamFactory(List<MemoryStream> pages)
-            {
-                _pages = pages;
-            }
-
-            public Stream CreatePageStream(int pageNumber)
+            switch (currentAngle)
             {
-                MemoryStream pageStream = new MemoryStream();
-
-                _pages.Add(pageStream);
-
-                return pageStream;
+                case 0:
+                    return postedAngle == 90 ? 90 : 270;
+                case 90:
+                    return postedAngle == 90 ? 180 : 0;
+                case 180:
+                    return postedAngle == 90 ? 270 : 90;
+                case 270:
+                    return postedAngle == 90 ? 0 : 180;
+                default:
+                    return 0;
             }
+        }
 
-            public void ReleasePageStream(int pageNumber, Stream pageStream)
+        /// <summary>
+        /// Gets enumeration member by rotation angle value.
+        /// </summary>
+        /// <param name="newAngle">New rotation angle value.</param>
+        /// <returns>Rotation enumeration member.</returns>
+        private static Rotation GetRotationByAngle(int newAngle)
+        {
+            switch (newAngle)
             {
-                //Do not release page stream as we'll need to keep the stream open
+                case 90:
+                    return Rotation.On90Degree;
+                case 180:
+                    return Rotation.On180Degree;
+                case 270:
+                    return Rotation.On270Degree;
+                default:
+                    return Rotation.On90Degree;
             }
         }
 
-        private static PageDescriptionEntity GetPageDescriptionEntities(Page page, string pagesInfoPath)
+        /// <summary>
+        /// Gets document pages data, dimensions and rotation angles.
+        /// </summary>
+        /// <param name="postedData">Posted data with document guid.</param>
+        /// <param name="loadAllPages">Flag to load all pages.</param>
+        /// <returns>Document pages data, dimensions and rotation angles.</returns>
+        private LoadDocumentEntity GetDocumentPages(PostedDataEntity postedData, bool loadAllPages)
```

Comment:
Codacy found an issue: [Make 'GetDocumentPages' a static method.](https://app.codacy.com/gh/groupdocs-viewer/GroupDocs.Viewer-for-.NET-MVC/pullRequest?prid=5850423)